src/test/java/jaiden/task/TaskListTest.java

```diff
@@ -0,0 +1,101 @@
+        tasks.add(new Deadline("test", true, LocalDate.parse("2025-08-22")));
+        tasks.add(new Event("test", true, LocalDate.parse("2025-08-22"), LocalDate.parse("2025-08-22")));
+        TaskList test = new TaskList(tasks);
+        assertEquals("     OK, I've marked this task as not done yet:\n       [T][ ] test\n", test.unmark(0));
+        assertEquals("     OK, I've marked this task as not done yet:\n       [D][ ] test (by: Aug 22 2025)\n", test.unmark(1));
+        assertEquals("     OK, I've marked this task as not done yet:\n       [E][ ] test (from: Aug 22 2025 to: Aug 22 2025)\n", test.unmark(2));
+    }
+
+    @Test
+    public void addTest() {
+        TaskList test = new TaskList();
+        assertEquals("     Got it. I've added this task:\n       [T][ ] test\n     Now you have 1 tasks in the list.\n", test.add(new Todo("test")));
+        assertEquals("     Got it. I've added this task:\n       [D][ ] test (by: Aug 22 2025)\n     Now you have 2 tasks in the list.\n", test.add(new Deadline("test", LocalDate.parse("2025-08-22"))));
+        assertEquals("     Got it. I've added this task:\n       [E][ ] test (from: Aug 22 2025 to: Aug 22 2025)\n     Now you have 3 tasks in the list.\n", test.add(new Event("test", LocalDate.parse("2025-08-22"), LocalDate.parse("2025-08-22"))));
+    }
+
+    @Test
+    public void removeTest() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        tasks.add(new Todo("test"));
+        tasks.add(new Deadline("test", LocalDate.parse("2025-08-22")));
+        tasks.add(new Event("test", LocalDate.parse("2025-08-22"), LocalDate.parse("2025-08-22")));
+        TaskList test = new TaskList(tasks);
+        assertEquals("     Noted. I've removed this task:\n       [E][ ] test (from: Aug 22 2025 to: Aug 22 2025)\n     Now you have 2 tasks in the list.\n", test.remove(2));
+        assertEquals("     Noted. I've removed this task:\n       [D][ ] test (by: Aug 22 2025)\n     Now you have 1 tasks in the list.\n", test.remove(1));
+        assertEquals("     Noted. I've removed this task:\n       [T][ ] test\n     Now you have 0 tasks in the list.\n", test.remove(0));
+    }
+
+    @Test
+    public void showTest() {
+        assertEquals("     Here are the tasks on Aug 22 2025 in your list:\n", new TaskList().show(LocalDate.parse("2025-08-22")));
+        ArrayList<Task> tasks = new ArrayList<>();
+        tasks.add(new Deadline("test", LocalDate.parse("2025-08-22")));
+        tasks.add(new Event("test", LocalDate.parse("2025-08-22"), LocalDate.parse("2025-08-22")));
+        tasks.add(new Deadline("test", LocalDate.parse("2024-08-22")));
+        tasks.add(new Event("test", LocalDate.parse("2024-08-22"), LocalDate.parse("2024-08-22")));
+        assertEquals("     Here are the tasks on Aug 22 2025 in your list:\n     1.[D][ ] test (by: Aug 22 2025)\n     2.[E][ ] test (from: Aug 22 2025 to: Aug 22 2025)\n", new TaskList(tasks).show(LocalDate.parse("2025-08-22")));
```

Comment:
Maybe you could consider splitting this line into two separate lines with the appropriate indentation level? 
```suggestion
        assertEquals("     Here are the tasks on Aug 22 2025 in your list:\n     1.[D][ ] test (by: Aug 22 2025)\n     2.[E][ ] test (from: Aug 22 2025 to: Aug 22 2025)\n", new TaskList(tasks).show(LocalDate.parse("2025-08-22")));
```